### PR TITLE
Working VAE Implementation

### LINKED ISSUE #10: Overfit Model on a Single Sample
Following #9 - another sanity check on real data.
Verify reconstruction is accurate and sharp.

### LINKED ISSUE #14: Plot Loss and Other Metrics During Training

### LINKED ISSUE #21: Rebuild VAE Using Keras Functional API
Building the model by extending `tf.keras.Model` disables certain useful features (e.g. `model.save()`, which includes information like the optimizer state and tensor names).

Replacing the class-based model definition with the Keras Functional API (via a `build_model()` function) is a more up-to-date approach

### PR DESCRIPTION
**Pull request by tasercake**
Fixes #21 
Fixes #10 
Fixes #14

### CODE REVIEW

**Comment by tasercake** [comment]
Fixes #21 
Fixes #10
